### PR TITLE
Delete unnecessary content in azure security keyvault jca

### PR DESCRIPTION
**Pull request by chenrujun**
Delete unnecessary content in azure security keyvault jca

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-jca/pom.xml [hunk 1]
```diff
@@ -17,26 +17,76 @@
...
+
     <build>
         <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.8.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-compiler-plugin;external_dependency} -->
-                <configuration>
-                    <source>1.8</source>
-                    <target>1.8</target>
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/pom.xml:28]: We will compile for both java 8 and java 11.

File: sdk/keyvault/azure-security-keyvault-jca/pom.xml [hunk 2]
```diff
@@ -17,26 +17,76 @@
...
-                    <encoding>${project.build.sourceEncoding}</encoding>
-                </configuration>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-resources-plugin</artifactId>
-                <version>2.4.3</version> <!-- {x-version-update;org.apache.maven.plugins:maven-resources-plugin;external_dependency} -->
-                <configuration>
-                    <encoding>${project.build.sourceEncoding}</encoding>
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/pom.xml:37]: The default value is `project.build.sourceEncoding`, we can delete configuration if we use the default velue.

Refs: https://maven.apache.org/plugins/maven-resources-plugin/resources-mojo.html

File: sdk/keyvault/azure-security-keyvault-jca/pom.xml [hunk 3]
```diff
@@ -17,26 +17,76 @@
     <version>2.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-security-keyvault-jca;current} -->
     <name>JCA Provider for Azure Key Vault</name>
     <description>The Java Crypto Architecture (JCA) Provider for Azure Key Vault</description>
+
+    <properties>
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/pom.xml:21]: From line 20 to 87: no logic change, just change order.

**Comment by chenrujun** [comment]
Hi, @yiliuTo , @zhichengliu12581 , please help to review this PR.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**